### PR TITLE
Mocha tests

### PR DESCRIPTION
**Pull request by eric79**
This branch creates a `mocha.js` driver to adding automated tests to a VWF app.  When added to the app's config file, it looks for a `test/test.js` file in the app and runs the tests that it finds there and displays the results in a poorly styled dialog.  :smile: 

There is a corresponding branch on the app side named `mocha-test` with a trivial test that always passes.

### CODE REVIEW

File: support/client/lib/vwf/view/mocha.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    return view.load( module, {
+
+        // == Module Definition ====================================================================
+
+        // -- initialize ---------------------------------------------------------------------------
+
+        initialize: function() {
+            mocha.setup( "bdd" );
+            window.assert = chai.assert;
+            $( "body" ).append( "<script src='test/test.js'></script>" ).append( " \
```
Comment (davideaster) [inline support/client/lib/vwf/view/mocha.js]: You can also add the Mocha stylesheets here as `<link rel='stylesheet' href='vwf/view/mocha/mocha.css'>`

File: support/client/lib/vwf/view/mocha.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                        <div class='modal-content'> \
+                            <div class='modal-header'> \
+                                <h4 class='modal-title'>Test results</h4> \
+                            </div> \
+                            <div class='modal-body'><div id='mocha'></div></div> \
+                        </div><!-- /.modal-content --> \
+                    </div><!-- /.modal-dialog --> \
+                </div><!-- /.modal --> \
+            " );
+            mocha.run();
```
Comment (davideaster) [inline support/client/lib/vwf/view/mocha.js]: This needs to be wrapped in `require` as:

    require( [ "test/test" ], function() { mocha.run() } );

The code in a dynamically created `<script/>` element isn't synchronous with its caller, so the tests generally won't be registered by the time `mocha.run` attempts to run them.

An added benefit is that this makes the test script a module that can depend on other files to help organize the tests into groups.

File: support/client/lib/vwf/view/mocha.js (hunk 1 above)
Comment (davideaster) [inline support/client/lib/vwf/view/mocha.js]: This `<script/>` can be omitted if the tests are loaded using `require`.

**Review by davideaster — COMMENTED** [review]
Good start! I put an example of how my suggestions would work for `/index.vwf` in commit d21f2816.

**Comment by eric79** [comment]
@davideaster , thanks for the comments and the example.  This is working much better now.

**Comment by landersk61** [comment]
:+1:

**Pull request merged**